app/controllers/security_group_controller.rb

```diff
@@ -193,28 +209,84 @@ def update
+
+    if session[:flash_msgs]
+      @flash_array = session[:flash_msgs].dup
+      session[:flash_msgs] = nil
+    end
+
+    td = session[:security_group][:task]
+    task = MiqTask.find(td[:id])
     if MiqTask.status_ok?(task.status)
-      add_flash(_("Security Group \"%{name}\" updated") % { :name => security_group_name })
+      add_flash(_("#{td[:resource]} #{td[:action]}d"))
     else
-      add_flash(_("Unable to update Security Group \"%{name}\": %{details}") % {
-        :name    => security_group_name,
+      add_flash(_("Unable to #{td[:action]} #{td[:resource]}: %{details}") % {
         :details => task.message
       }, :error)
     end
 
-    @breadcrumbs.pop if @breadcrumbs
-    session[:edit] = nil
     session[:flash_msgs] = @flash_array.dup if @flash_array
 
-    javascript_redirect :action => "show", :id => security_group_id
+    if !session[:security_group][:tasks].empty?
+      task = session[:security_group][:tasks].shift
+      session[:security_group][:task] = task
+      initiate_wait_for_task(:task_id => task[:id], :action => "update_finished")
+    else
+      @breadcrumbs.pop if @breadcrumbs
+      session[:edit] = nil
+      javascript_redirect :action => "show", :id => security_group_id
+    end
   end
 
   private
 
+  def create_rule(rule)
+    rule_data = @security_group.class.parse_security_group_rule(rule)
+    task_id = @security_group.create_security_group_rule_queue(session[:userid], @security_group.ems_ref,
+                                                               rule["direction"], rule_data)
+    if task_started(task_id, "Security Group Rule Create")
+      @tasks << {:id => task_id, :resource => "Security Group #{@security_group.name} Update: Rule", :action => :create}
+    end
+  end
+
+  def delete_rule(id)
+    task_id = @security_group.delete_security_group_rule_queue(session[:userid], id)
+    if task_started(task_id, "Security Group Rule Delete")
+      @tasks << {:id => task_id, :resource => "Security Group #{@security_group.name} Update: Rule", :action => :delete}
+    end
+  end
+
+  def rule_changed?(params)
+    sg_rule = @security_group.firewall_rules.find(params["id"])
+    return true if params["direction"] && sg_rule.direction != params["direction"]
+    return true if params["end_port"] && sg_rule.end_port != params["end_port"]
+    return true if params["host_protocol"] && sg_rule.host_protocol != params["host_protocol"]
+    return true if params["network_protocol"] && sg_rule.network_protocol != params["network_protocol"]
+    return true if params["port"] && sg_rule.port != params["port"]
+    if params["source_ip_range"] && !params["source_ip_range"].empty? &&
+       sg_rule.source_ip_range != params["source_ip_range"]
+      return true
+    end
+    if params["source_security_group_id"] && !params["source_security_group_id"].empty? &&
+       sg_rule.source_security_group_id != params["source_security_group_id"]
+      return true
+    end
+  end
```

Comment:
I wonder if the `rule_changed?` method could be written differently...
Currently all conditionals return `true`, and looks like a `false` is never returned.
Would it be simpler and shorter to write a conditional that returns a `false` instead of listing out all the conditionals that return a `true`?

I'm also trying to target the cyclomatic complexity CodeClimate error here by reducing the number of conditionals.